Sources/LeafKit/LeafParser.swift

```diff
@@ -1,3 +1,76 @@
+extension String: Error {}
```

Comment:
We should probably be using `LeafError` even during development, as passing around the source location info can be tricky but is necessary to give useful error message.

Reply:
I'll add that in for the parsers, this is mostly a temporary thing, but it helps dev.. will keep internal